src/services/ui/src/pages/actions/WithdrawPackage.tsx

```diff
@@ -0,0 +1,261 @@
+import { Navigate, useNavigate, useParams } from "react-router-dom";
+import { Button } from "@/components/Inputs";
+import { ConfirmationModal } from "@/components/Modal/ConfirmationModal";
+import { useState } from "react";
+import { Action, ItemResult, withdrawPackageSchema } from "shared-types";
+import { FAQ_TARGET, ROUTES } from "@/routes";
+import { PackageActionForm } from "./PackageActionForm";
+import { ActionFormIntro, PackageInfo } from "./common";
+import { z } from "zod";
+import { SubmitHandler, useForm } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import * as I from "@/components/Inputs";
+import { Link } from "react-router-dom";
+import { LoadingSpinner } from "@/components";
+import { buildActionUrl } from "@/lib";
+import { useGetUser } from "@/api/useGetUser";
+import { submit } from "@/api/submissionService";
+
+const withdrawPackageFormSchema = withdrawPackageSchema(
+  z.array(z.instanceof(File))
+);
+type WithdrawPackageFormSchema = z.infer<typeof withdrawPackageFormSchema>;
+type UploadKey = keyof WithdrawPackageFormSchema["attachments"];
+type AttachmentRecipe = {
+  readonly name: UploadKey;
+  readonly label: string;
+  readonly required: boolean;
+};
+
+const attachments: AttachmentRecipe[] = [
+  {
+    name: "supportingDocumentation",
+    label: "Supporting Documentation",
+    required: false,
+  } as const,
+];
+
+const WithdrawPackageForm: React.FC = ({ item }: { item?: ItemResult }) => {
+  const [successModalIsOpen, setSuccessModalIsOpen] = useState(false);
+  const [errorModalIsOpen, setErrorModalIsOpen] = useState(false);
+  const [cancelModalIsOpen, setCancelModalIsOpen] = useState(false);
+  const navigate = useNavigate();
+  const { id, type } = useParams<{ id: string; type: Action }>();
+  const { data: user } = useGetUser();
+  const form = useForm<WithdrawPackageFormSchema>({
+    resolver: zodResolver(withdrawPackageFormSchema),
+  });
+
+  const handleSubmit: SubmitHandler<WithdrawPackageFormSchema> = async (
+    data
+  ) => {
+    try {
+      await submit<WithdrawPackageFormSchema>({
```

Comment:
@13bfrancis So, when `data` comes thru, is it right to assume it's been validated using the Zod resolver?? Because if so, it _should_ fail, right? We don't give it an `id` but that's a part of the main schema... 🤔 

I'd taken the the approach of chaining the `id` field onto the Zod type for IssueRai and some others: https://github.com/Enterprise-CMCS/macpro-mako/blob/4f38d73c3794f9747d23274149f78cb7b7b52b3e/src/services/ui/src/pages/actions/IssueRai.tsx#L79-L94